Dockerfile

```diff
@@ -1,28 +1,33 @@
 FROM quay.io/centos/centos:stream8
 
 RUN dnf -y module install python39 && dnf -y install python39 python39-pip
-RUN mkdir /app
-ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app
-ADD example_plugin.py /app
-ADD test_example_plugin.py /app
-ADD requirements.txt /app
+RUN mkdir /app \
+ && chmod 700 /app
+ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app/
+COPY README.md /app/
+COPY poetry.lock /app/
+COPY pyproject.toml /app/
+COPY example_plugin.py /app/
+COPY test_example_plugin.py /app/
 WORKDIR /app
 
-RUN pip3 install -r requirements.txt
+RUN python3.9 -m pip install poetry \
```

Comment:
We don't need to do it, but it may help to codify the poetry version with "poetry>=1.2.2,<1.3"
That would make sure it's the newest version of poetry 1.2, which will help ensure it works.

Reply:
Not to pile on, but we might want to consider using `poetry` outside the container to generate a `requirements.txt` file that can be used by `pip3` to install the right modules.

Such a change would result in lines 14 - 17 being removed, and line 19 becoming something like:

```
 && python3.9 -m pip install coverage -r requirements.txt \
```

Reply:
I think we should take a fair look at this recommendation from Peter before pushing this forward. It makes sense to me to simplify the container and remove the poetry bits and actions from there.

Reply:
How should we automate the generation of the requirements.txt? Would it be in another build stage in the Dockerfile?